pkg/operator/controller/gatewayapi/controller.go

```diff
@@ -84,16 +79,7 @@ type reconciler struct {
 func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	log.Info("reconciling", "request", request)
 
-	var fg configv1.FeatureGate
-	if err := r.cache.Get(ctx, request.NamespacedName, &fg); err != nil {
-		if apierrors.IsNotFound(err) {
-			log.Info("featuregate not found; reconciliation will be skipped", "request", request)
-			return reconcile.Result{}, nil
-		}
-		return reconcile.Result{}, err
-	}
-
-	if !featureIsEnabled(featureGateName, &fg) {
+	if !r.config.GatewayAPIEnabled {
```

Comment:
With this change, there isn't really a good reason for this controller to watch featuregates at all.  However, it does make sense to have a control loop to retry creating a CRD when a create fails.  In addition, we really do want to have the controller but have it reconcile customresourcedefinitions if they change (see <https://issues.redhat.com/browse/NE-1273>), so let's leave it as a followup to modify the controller to watch and reconcile customresourcedefinitions.